pkg/controller/container-runtime-config/helpers.go

```diff
@@ -196,7 +209,7 @@ func updateStorageConfig(data []byte, internal *mcfgv1.ContainerRuntimeConfigura
 		return nil, fmt.Errorf("error decoding crio config: %v", err)
 	}
 
-	if internal.OverlaySize != (resource.Quantity{}) {
+	if internal.OverlaySize.Value() != 0 {
```

Comment:
Is this related to the topic of this PR? And how does it related to current line 262 in this file (`LogSizeMax`) checking for both?

I would also expect the relationship of conditions in `Controller.syncRuntimeConfig` and {`updateStorageConfig` vs. `createCRIODropInFiles`} to be consistent — either the caller and the callee should be checking for exactly the same condition, or the caller should be checking for “is the value present” and the callee for “is the value (which is known to be present) something that requires a config change”.

Right now `OverlaySize` and `LogSizeMax` are treated differently and I’m not sure whether there is some hard-to-notice reason for that.

Reply:
Fixed